blockchain/src/state/read/response/conversions.rs

```diff
@@ -0,0 +1,298 @@
+        MoveStructTag {
+            address: *self.address_identifier_at(m_handle.address),
+            module: self.identifier_at(m_handle.name).as_str().into(),
+            name: self.identifier_at(s_handle.name).as_str().into(),
+            generic_type_params: type_params.iter().map(|t| self.new_move_type(t)).collect(),
+        }
+    }
+
+    fn new_move_type(&self, token: &SignatureToken) -> MoveType {
+        match token {
+            SignatureToken::Bool => MoveType::Bool,
+            SignatureToken::U8 => MoveType::U8,
+            SignatureToken::U16 => MoveType::U16,
+            SignatureToken::U32 => MoveType::U32,
+            SignatureToken::U64 => MoveType::U64,
+            SignatureToken::U128 => MoveType::U128,
+            SignatureToken::U256 => MoveType::U256,
+            SignatureToken::Address => MoveType::Address,
+            SignatureToken::Signer => MoveType::Signer,
+            SignatureToken::Vector(t) => MoveType::Vector {
+                items: Box::new(self.new_move_type(t.borrow())),
+            },
+            SignatureToken::Struct(v) => MoveType::Struct(self.new_move_struct_tag(v, &[])),
+            SignatureToken::StructInstantiation(shi, type_params) => {
+                MoveType::Struct(self.new_move_struct_tag(shi, type_params))
+            }
+            SignatureToken::TypeParameter(i) => MoveType::GenericTypeParam { index: *i },
+            SignatureToken::Reference(t) => MoveType::Reference {
+                mutable: false,
+                to: Box::new(self.new_move_type(t.borrow())),
+            },
+            SignatureToken::MutableReference(t) => MoveType::Reference {
+                mutable: true,
+                to: Box::new(self.new_move_type(t.borrow())),
+            },
+            SignatureToken::Function(args, result, abilities) => {
+                let new_vec = |toks: &[SignatureToken]| {
+                    toks.iter()
+                        .map(|t| self.new_move_type(t))
+                        .collect::<Vec<_>>()
+                };
+                MoveType::Function {
+                    args: new_vec(args),
+                    results: new_vec(result),
+                    abilities: *abilities,
+                }
+            }
+        }
+    }
+
+    fn new_move_struct(&self, def: &StructDefinition) -> MoveStruct {
+        let handle = self.struct_handle_at(def.struct_handle);
+        let (is_native, fields) = match &def.field_information {
+            StructFieldInformation::Native => (true, vec![]),
+            StructFieldInformation::Declared(fields) => (
+                false,
+                fields
+                    .iter()
+                    .map(|f| self.new_move_struct_field(f))
+                    .collect(),
+            ),
+            StructFieldInformation::DeclaredVariants(..) => {
+                // TODO(#13806): implement for enums. Currently we pretend they don't have fields
```

Comment:
This todo is referencing an aptos issue. Is there a way we can avoid inheriting their tech debt?